### PR TITLE
Use CBA Versioning System properly

### PR DESCRIPTION
**Pull request by jonpas**
**When merged this pull request will:**
- Title

**Merge after https://github.com/CBATeam/CBA_A3/pull/1221 is released!**

TODO:
- [x] Copy updated `script_macros_common.hpp` on CBA release
  - With cleaned-up quoting `##`

### CODE REVIEW

**Comment by jonpas** [comment]
CBA released, this is done and ready for merge.

**Comment by mharis001** [comment]
Should this bump `REQUIRED_CBA_VERSION`?

**Comment by jonpas** [comment]
No need, if older CBA version is used with this it'll just show the short version (`v3.12`) instead of the full one (`v3.12.6.43`) in the map "CBA > Credits" section.

File: include/x/cba/addons/main/script_macros_common.hpp [hunk 1]
```diff
@@ -785,13 +819,13 @@ Macro: ISNILS()
...
-// #define PREP_SYS2(var1,var2,var3,var4) ##var1##_##var2##_fnc_##var4 = { ##var1##_##var2##_fnc_##var4 = COMPILE_FILE_SYS(var1,var3,DOUBLES(fnc,var4)); if (isNil "_this") then { call ##var1##_##var2##_fnc_##var4 } else { _this call ##var1##_##var2##_fnc_##var4 } }
+// #define PREPMAIN_SYS(var1,var2,var3) var1##_fnc_##var3 = { var1##_fnc_##var3 = COMPILE_FILE_SYS(var1,var2,DOUBLES(fnc,var3)); if (isNil "_this") then { call var1##_fnc_##var3 } else { _this call var1##_fnc_##var3 } }
+// #define PREP_SYS(var1,var2,var3) var1##_##var2##_fnc_##var3 = { var1##_##var2##_fnc_##var3 = COMPILE_FILE_SYS(var1,var2,DOUBLES(fnc,var3)); if (isNil "_this") then { call var1##_##var2##_fnc_##var3 } else { _this call var1##_##var2##_fnc_##var3 } }
+// #define PREP_SYS2(var1,var2,var3,var4) var1##_##var2##_fnc_##var4 = { var1##_##var2##_fnc_##var4 = COMPILE_FILE_SYS(var1,var3,DOUBLES(fnc,var4)); if (isNil "_this") then { call var1##_##var2##_fnc_##var4 } else { _this call var1##_##var2##_fnc_##var4 } }
```
Comment (CreepPork) [inline include/x/cba/addons/main/script_macros_common.hpp]: I'm a bit lost, why are these macros here but commented out?

**Pull request merged**